### DISCUSSION TITLE
How can make <leader>e only focus the file explorer (NeoTree) if it's already open ?

### DISCUSSION THREAD

**Discussion by cassepipe** [nvim-neo-tree/neo-tree.nvim#1566] [Q&A]
How can make <leader>e only focus the file explorer (NeoTree) if it's already open ?

**Comment by miversen33** [comment]
There isn't an option/configuration you can provided to do that kind of behavior (as far as I know). You would need to craft it yourself using [:h neo-tree-events](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/main/doc/neo-tree.txt#L1375). Basically you want to hook 2 events

- `neo_tree_window_after_open`
- `neo_tree_window_before_close`

In your `after_open` event, you would register a keymap that sets `<Leader>e` to focus on the neo-tree buffer (there may be a function in neo-tree you can call to steal focus, I don't remember).

In your `before_close` event, you would unregister that keymap.
